lib/fluent/plugin/mongo_auth.rb

```diff
@@ -8,23 +8,25 @@ def self.included(klass)
         config_param :password, :string, default: nil, secret: true
         desc "MongoDB authentication database"
         config_param :auth_source, :string, default: nil
+        desc "MongoDB authentication mechanism"
```

Comment:
Can we add `auth_mech` validity checking code in configure?

Reply:
If mongo driver raises a good error with good message, we can delegate its validation to mongo driver. If not, we should add the validation or use `enum` config type for it.

Reply:
Mongo Ruby driver seems to generate a good error message with this line:
https://github.com/mongodb/mongo-ruby-driver/blob/584ba3869e6288aa5b7835bd2281bb10fd692f78/lib/mongo/uri.rb#L631-L633
We can delegate its validation into Mongo Ruby driver, I think.

Reply:
I see but this seems to log only warning message, not raising an error.
I think auth_mech is for security so `Fluent::ConfigError` should be raised for invalid parameter.

Reply:
It seems this doesn't get checked until an action is performed that requires authentication. I've added a check to the plugin.